zebedee-cms/src/main/java/com/github/onsdigital/zebedee/util/slack/SlackNotifier.java

```diff
@@ -32,4 +50,70 @@ public void collectionAlarm(Collection c, String alarm, PostMessageField... args
         SlackNotification.alarm(alarm, args);
     }
+
+
+    @Override
+    public void sendSlackMessage(PostMessage message) throws Exception{
+        try {
+            slackClient.sendMessage(message);
+        } catch (Exception ex) {
+            throw new Exception("unexpected error sending slack message", ex);
+        }
+    }
+
+    @Override
+    public PostMessage createPostMessage(String channel, String text) {
+        return profile.newPostMessage(channel, text);
+    }
+
+    @Override
+    public void callCollectionAlarm(Collection collection, String channel, String customMessage, Exception ex) {
+        AttachmentField exField = new AttachmentField("exception", ex.getMessage(), false);
+        PostMessage postMessage = createPostMessage(channel, customMessage)
+                .addAttachment(createCollectionAttachment("Alert", "Collection Alarm", Colour.DANGER, collection, exField));
+
+        try {
+            sendSlackMessage(postMessage);
+        } catch (Exception e) {
+            error().exception(e).log("unexpected error while sending slack notification");
+        }
+    }
+
+    @Override
+    public void callCollectionWarning(Collection collection, String channel, String customMessage, AttachmentField... fields) {
+        PostMessage postMessage = createPostMessage(channel, customMessage)
+                .addAttachment(createCollectionAttachment("Warning", "Collection Warning", Colour.WARNING, collection, fields));
+        try {
+            sendSlackMessage(postMessage);
+        } catch (Exception e) {
+            error().exception(e).log("unexpected error while sending slack notification");
+        }
+    }
+
+    @Override
+    public void callCollectionAlarm(Collection collection, String channel, String customMessage, AttachmentField... fields) {
+        PostMessage postMessage = createPostMessage(channel, customMessage)
+                .addAttachment(createCollectionAttachment("Alert", "Collection Alarm", Colour.DANGER, collection, fields));
+        try {
+            sendSlackMessage(postMessage);
+        } catch (Exception e) {
+            error().exception(e).log("unexpected error while sending slack notification");
+        }
+    }
+
+    private PostMessageAttachment createCollectionAttachment(String title, String message, Colour colour, Collection collection, AttachmentField... fields) {
+        PostMessageAttachment attachment = new PostMessageAttachment(title, message, colour)
+                .addField("Publishing Type", collection.getDescription().getType().name(), true)
+                .addField("CollectionID", collection.getId(), false)
+                .addField("Collection Name", collection.getDescription().getName(), false);
+
+        if (fields != null)  {
+            for(AttachmentField field : fields) {
+                attachment.addField(field.getTitle(), field.getMessage(), field.isShort());
+            }
+        }
+
+        return attachment;
```

Comment:
You also need to add tests for the new functionality you've added to this class.